### PR TITLE
HLLs extracted for search

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
- extract hyperloglog(set) datatypes to search on hll values, e.g. range (similar to counters)
- update default schema for hll(s)
- rebar.config (for now) set to feature-zl-hll_datatypes of riak_kv
  until reviewed/merged correctly
- add r_t and eunit tests respectively

### CODE REVIEW

**Comment by russelldb** [comment]
Please, how do I run the riak_test? Just build yoko branch and run riak_test?

**Comment by russelldb** [comment]
Ah, s'ok, I found the task in the Makefile

File: riak_test/yz_dt_test.erl [hunk 1]
```diff
@@ -71,6 +75,34 @@ set_search(Node) ->
...
 
+hll_update(PB) ->
+    Dynamos = lists:foldl(fun riakc_hll:add_element/2, riakc_hll:new(),
+                          [<<"Riak">>, <<"Cassandra">>, <<"Voldemort">>,
+                           <<"Couchbase">>]),
+    Erlangs = riakc_hll:add_elements([<<"Riak">>,
+                                      <<"Couchbase">>,
+                                      <<"CouchDB">>],
+                                     riakc_hll:new()),
+    Bucket = {?HLL, <<"databass">>},
```
Comment (russelldb) [inline riak_test/yz_dt_test.erl:86]: typo? ;)

**Comment by russelldb** [comment]
Test pass for me, code looks sane

+1 from me, would like a second opinion from @fadushin please?

File: riak_test/yz_dt_test.erl (hunk 1 above)
Comment (zeeshanlakhani) [inline riak_test/yz_dt_test.erl:86]: ;)

File: src/yz_extractor.erl [hunk 2]
```diff
@@ -90,7 +91,7 @@ get_map() ->
     case mochiglobal:get(?META_EXTRACTOR_MAP, undefined) of
         undefined ->
             Map = get_map_read_through(),
-            mochiglobal:put(?META_EXTRACTOR_MAP, Map),
+             mochiglobal:put(?META_EXTRACTOR_MAP, Map),
```
Comment (jvoegele) [inline src/yz_extractor.erl]: Extra space of indentation here?

Comment (zeeshanlakhani) [inline src/yz_extractor.erl]: good catch. Fixed push once review is complete.

File: test/yz_dt_extractor_tests.erl [hunk 3]
```diff
@@ -82,4 +90,10 @@ raw_type(set) ->
       );
 raw_type(counter) ->
     ?COUNTER_TYPE(
-       element(2,?COUNTER_TYPE:update({increment, 10}, <<0>>, ?COUNTER_TYPE:new()))).
+       element(2,?COUNTER_TYPE:update({increment, 10}, <<0>>, ?COUNTER_TYPE:new())));
+raw_type(hll) ->
+    ?HLL_TYPE(
+       element(2,?HLL_TYPE:update({add_all, [<<"T">>, <<"h">>, <<"r">>, <<"i">>,
+                                             <<"l">>, <<"l">>, <<"a">>, <<"s">>,
+                                             <<"f">>, <<"u">>, <<"h">>]},
```
Comment (jvoegele) [inline test/yz_dt_extractor_tests.erl:98]: What is the significance of the above list of binaries containing seemingly arbitrary letters?

Comment (zeeshanlakhani) [inline test/yz_dt_extractor_tests.erl:98]: @jvoegele just using unique (and non-unique) simple binaries for the test. See the <<"l">> for example. It also spells "Thrill as fuh" and is a play on Andy Pavlo's Trill as f**\* post... jokes are better explained.

**Comment by jvoegele** [comment]
LGTM. I can `+1` after @zeeshanlakhani pushes the fix for the minor formatting issue.

**Comment by zeeshanlakhani** [comment]
@jvoegele pushed and did a rebase since it's just a "messup" commit w/ spacing since you're ready to +1 anyhow :).

**Comment by jvoegele** [comment]
+1 f8dff25

File: rebar.config [hunk 4]
```diff
@@ -13,7 +13,7 @@
   {kvc, ".*",
    {git, "git://github.com/etrepum/kvc.git", {tag, "v1.5.0"}}},
   {riak_kv, ".*",
-   {git, "git://github.com/basho/riak_kv.git", {branch, "develop-2.2"}}},
+   {git, "git://github.com/basho/riak_kv.git", {branch, "feature-zl-hll_datatypes"}}},
```
Comment (fadushin) [inline rebar.config]: Should this be reverted back to develop-2.2?

Comment (zeeshanlakhani) [reply rebar.config]: good point. Will update after review. Not sure how that changed.

Comment (zeeshanlakhani) [reply rebar.config]: @fadushin anything else?

**Comment by fadushin** [comment]
+1 b35204d

**Comment by fadushin** [comment]
@borshop merge

**Pull request merged**